src/state.rs

```diff
@@ -114,10 +123,8 @@ impl State {
             && (self.consensus != ConsensusType::Delegated || self.validator_set.len() < 1)
         {
             self.validator_set.push(Validator {
-                subnet: SubnetID::new(&self.parent_id, Address::new_id(sdk::message::receiver())),
                 addr: addr.clone(),
-                // FIXME: Receive address in params
-                net_addr: String::new(),
+                net_addr: String::from(net_addr),
```

Comment:
We should consider checking if this is a multiaddr. We should only support multiaddr for this. We can add a TODO and then consider it for the integration.